30-remote-site-setup.sh

```diff
@@ -35,6 +35,37 @@ EOF
   ls -l "$ssh_dir"
 }
 
+# Populate the bosco override dir from a Git repo
+# Expected Git repo layout:
+#     RESOURCE_NAME_1/
+#         bosco_override/
+#         ...
+#     RESOURCE_NAME_2/
+#         bosco_override/
+#         ...
+#     ...
+function bosco_override_setup {
+    GIT_ENDPOINT=$1
+    RESOURCE_NAME=$2
+
+    # Set up config for SSH endpoints
+    if [[ $GIT_ENDPOINT =~ ^[A-Za-z0-9_-]+@([^:]+): ]]; then
+        ssh-keyscan "${BASH_REMATCH[1]}" >> ~/.ssh/known_hosts
```

Comment:
```suggestion
        grep -Fqs "${BASH_REMATCH[1]}" ~/.ssh/known_hosts || ssh-keyscan "${BASH_REMATCH[1]}" >> ~/.ssh/known_hosts
```
check if the host is already in the known_hosts file to avoid creating duplicate entries.

Reply:
Does a duplicate really matter?

Reply:
Actually, it might if the hostkey changes.

Reply:
But if the hostkey changes and we keep the old one, we're screwed regardless

Reply:
If we blindly accept the hostkey whenever it changes, it defeats the whole point of hostkeys.

Then again,
- times hostkey change has saved me from MITM attack: 0
- times hostkey change has annoyed the crap out of me: 123921497

Reply:
P.S.: Don't show the above comment to the sec team or they'll make it their next exercise :-P